spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -1498,7 +1498,13 @@ protected void handleAsyncFailure() {
 			// We will give up on retrying with the remaining copied and failed Records.
 			for (FailedRecordTuple<K, V> copyFailedRecord : copyFailedRecords) {
 				try {
-					invokeErrorHandlerBySingleRecord(copyFailedRecord);
+						KafkaListenerObservation.LISTENER_OBSERVATION.observation(
+										getContainerProperties().getObservationConvention(),
+										DefaultKafkaListenerObservationConvention.INSTANCE,
+										() -> new KafkaRecordReceiverContext(copyFailedRecord.record(), getListenerId(),
+												getClientId(), this.consumerGroupId, this::clusterId),
+										this.observationRegistry)
```

Comment:
I don't think that it is OK to start a new observation.
The error handling is really a part of already started one.
I will dig later a bit more how we do that in other places.